org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/WorkDoneProgressKind.java

```diff
@@ -0,0 +1,43 @@
+/******************************************************************************
+ * Copyright (c) 2016 TypeFox and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ ******************************************************************************/
+package org.eclipse.lsp4j;
+
+public enum WorkDoneProgressKind {
+	BEGIN("begin"),
+	REPORT("report"),
+	END("end");
+
+	private final String value;
+	
+	WorkDoneProgressKind(String value) {
+		this.value = value;
+	}
+	
+	public String getValue() {
+		return value;
+	}
+	
+	public static WorkDoneProgressKind forValue(String value) {
+		WorkDoneProgressKind[] allValues = WorkDoneProgressKind.values();
+		for (WorkDoneProgressKind workDoneProgressKind : allValues) {
+			if(workDoneProgressKind.getValue().equals(value)) {
+				return workDoneProgressKind;
+			}
+		}
+		throw new IllegalArgumentException("Illegal enum value: " + value);
+	}
+
+	@Override
+	public String toString() {
```

Comment:
Not sure about this. The doc allows to override toString() but it kind-of suggests that it should not be necessary:
```
Returns the name of this enum constant, as contained in the declaration. This method may be overridden, though it typically isn't necessary or desirable. An enum type should override this method when a more "programmer-friendly" string form exists.
```